Add workflows

ederic-oytas opened the pull request:
This pull request adds workflows to test the front-end and back-end for Tabby. For now, these workflows will run on pull request. The front-end workflow only lints for now, and the back-end workflow checks formatting and runs unit tests.

This pull request also configures the linter for the front-end.

David1000734 commented on server/tabby_server/__main__.py:
Does the lint from python want no spaces between the operations?
'methods = [...'
has to be replaced with
'methods=[...'

David1000734 commented on .github/workflows/workflow.yml:
We should consider caching the dependencies in the future to speed up workflow build times

David1000734 reviewed (approved):
Approve

ederic-oytas commented:
Apologies, it seems that my other commits were also added as well. I updated the README.md in server/ to include the instructions for installing poetry and using poetry install to install all of the dependencies.
